network/stream/peer.go

```diff
@@ -134,32 +134,29 @@ type want struct {
 	closeC    chan error          // signal polling goroutine to terminate due to empty batch or timeout
 }
 
-// getOfferOrDrop gets on open offer for the requested ruid
-// in case the offer is not found - the peer is dropped
-func (p *Peer) getOfferOrDrop(ruid uint) (o offer, shouldBreak bool) {
+// getOffer gets on open offer for the requested ruid
+// in case the offer is not found - error is returned
+func (p *Peer) getOffer(ruid uint) (o offer, err error) {
 	p.mtx.RLock()
 	o, ok := p.openOffers[ruid]
 	p.mtx.RUnlock()
 	if !ok {
-		p.logger.Error("ruid not found, dropping peer", "ruid", ruid)
-		p.Drop("ruid not found")
-		return o, true
+		return o, fmt.Errorf("ruid not found, dropping peer: %d", ruid)
```

Comment:
Maybe to remove `dropping` from the error message, as that consequence is handled in different place.